src/common/service/NetRemoteService.cxx

```diff
@@ -494,6 +620,116 @@ NetRemoteService::WifiAccessPointSetPhyType([[maybe_unused]] ::grpc::ServerConte
+        return handleFailure(WifiAccessPointOperationStatusCode::WifiAccessPointOperationStatusCodeInvalidParameter, "Invalid authentication algorithm provided");
+    } else if (!std::empty(cipherSuites) && std::ranges::find(cipherSuites, Dot11CipherSuite::Dot11CipherSuiteUnknown) != std::cend(cipherSuites)) {
+        return handleFailure(WifiAccessPointOperationStatusCode::WifiAccessPointOperationStatusCodeInvalidParameter, "Invalid cipher suite provided");
+    }
+
+    // Create an AP controller for the requested AP.
+    auto accessPointWeak = m_accessPointManager->GetAccessPoint(request->accesspointid());
+    auto accessPointController = detail::IAccessPointWeakToAccessPointController(accessPointWeak);
+    if (!accessPointController) {
+        return handleFailure(WifiAccessPointOperationStatusCode::WifiAccessPointOperationStatusCodeAccessPointInvalid, std::format("Failed to create controller for access point {}", request->accesspointid()));
+    }
+
+    // Convert Dot11AkmSuiteConfigurations to Ieee equivalent values.
+    std::vector<Microsoft::Net::Wifi::Ieee80211AkmSuite> ieeeAkmSuites(static_cast<std::size_t>(std::size(akmSuiteConfigurations)));
+    std::vector<Microsoft::Net::Wifi::Ieee80211AuthenticationAlgorithm> ieeeAuthenticationAlgorithms(static_cast<std::size_t>(std::size(akmSuiteConfigurations)));
+    std::ranges::transform(akmSuiteConfigurations | std::views::transform(detail::AkmSuiteFromDot11AkmSuiteConfiguration), std::begin(ieeeAkmSuites), detail::NetRemoteAkmToIeee80211AkmSuite);
+    std::ranges::transform(akmSuiteConfigurations | std::views::transform(detail::AuthenticationAlgorithmFromAkmSuiteConfiguration), std::begin(ieeeAuthenticationAlgorithms), detail::NetRemoteAuthenticationAlgorithmToIeeeAuthenticationAlgorithm);
+    // TODO: Store configuration values somehow.
+
+    // Convert Dot11CipherSuites to Ieee80211CipherSuites.
+    std::vector<Microsoft::Net::Wifi::Ieee80211CipherSuite> ieeeCipherSuites(static_cast<std::size_t>(std::size(cipherSuites)));
+    std::ranges::transform(cipherSuites, std::begin(ieeeCipherSuites), [&](const auto& cipherSuite) {
+        return detail::NetRemoteCipherSuiteToIeeeCipherAlgorithm(static_cast<Dot11CipherSuite>(cipherSuite));
+    });
+
+    // Check if provided authentication configuration is supported by AP.
+    try {
+        auto accessPointCapabilities = accessPointController->GetCapabilities();
+        const auto& supportedAkmSuites = accessPointCapabilities.AkmSuites;
+        const auto& supportedCipherSuites = accessPointCapabilities.CipherSuites;
+
+        if (!std::empty(akmSuiteConfigurations)) {
+            // Remove unsupported AKM suites.
+            std::erase_if(ieeeAkmSuites, [&](const auto& ieeeAkmSuite) {
+                return std::ranges::find(supportedAkmSuites, ieeeAkmSuite) == std::cend(supportedAkmSuites);
+            });
+            // TODO: Do the same for other parts of Dot11AkmSuiteConfiguration.

```

Comment:
Are there actually any parts left to do?

Reply:
Yes, we still need the configuration values (shared key, radius server configuration, or empty) that come from `oneof Configuration` in `WifiCore.proto`, which also need to be tightly coupled with the AKM suite. I'm working on some changes to do this, but I'm struggling a bit...